timesketch/lib/analyzers/login.py

```diff
@@ -0,0 +1,178 @@
+        return 0
+
+    attributes = {}
+    attributes['username'] = string_list[1]
+    attributes['logon_domain'] = string_list[2]
+    attributes['session_id'] = string_list[3]
+
+    logon_type_code = string_list[4]
+    attributes['logon_type'] = LOGON_TYPES.get(
+        logon_type_code, LOGON_TYPES.get(u'0'))
+
+    # Want to add an emoji in case this is a screensaver unlock.
+    if logon_type_code == '7':
+        screen_emoji = emojis.get_emoji('screen')
+        emojis_to_add.append(screen_emoji)
+
+    event.add_tags(['logoff-event'])
+    event.add_attributes(attributes)
+    event.add_emojis(emojis_to_add)
+    return 1
+
+
+def parse_evtx_logon_event(event, string_list, string_parsed, emoji):
+    """Parse logon events and return a count of event processed.
+
+    Args:
+        event: an event object (instance of interface.Event).
+        string_list: a list of strings extracted from the Event Log.
+        emoji: string with the Unicode point for the emoji that will be
+            added to the event.
+
+    Returns:
+        int: 1 if the event got processed correctly, 0 otherwise.
+    """
+    attributes = {}
+    emojis_to_add = [emoji]
+
+    if not string_parsed:
+        string_parsed = {}
+        if not len(string_list) == 23:
+            return 0
+        string_parsed['target_user_id'] = string_list[4]
+        string_parsed['target_user_name'] = string_list[5]
+        string_parsed['hostname'] = string_list[11]
+
+    logon_type_code = string_list[8]
+    attributes['logon_type'] = LOGON_TYPES.get(
+        logon_type_code, LOGON_TYPES.get(u'0'))
+
+    # Want to add an emoji in case this is a screensaver unlock.
+    if logon_type_code == '7':
+        screen_emoji = emojis.get_emoji('screen')
+        emojis_to_add.append(screen_emoji)
+
+    attributes['username'] = '{0:s}/{1:s}'.format(
+        string_parsed.get('target_user_id', 'N/A'),
+        string_parsed.get('target_user_name', 'Unknown'))
+    attributes['hostname'] = string_parsed.get('target_machine_name', 'N/A')
+    attributes['session_id'] = string_list[3]
+
+    event.add_tags(['logon-event'])
+    event.add_attributes(attributes)
+    event.add_emojis(emojis_to_add)
+    return 1
+
+
+class LoginSketchPlugin(interface.BaseSketchAnalyzer):
+    """Sketch analyzer for Login and Logoff related activity."""
+
+    NAME = 'login'
+    DEPENDENCIES = frozenset()
+
+    def __init__(self, index_name, sketch_id):
+        """Initialize The Sketch Analyzer.
+
+        Args:
+            index_name: Elasticsearch index name
+            sketch_id: Sketch ID
+        """
+        self.index_name = index_name
+        super(LoginSketchPlugin, self).__init__(index_name, sketch_id)
+
+    def run(self):
+        """Entry point for the analyzer.
+
+        Returns:
+            String with summary of the analyzer result
+        """
+        screenshot_emoji = emojis.get_emoji('camera')
+        login_emoji = emojis.get_emoji('lock')
```

Comment:
Shouldn't "login" be unlock and logout be "lock"?

Reply:
yes it should... good catch... this was an honest mistake on my behalf ;)